OpenRA.Mods.Common/Effects/Shrapnel.cs

```diff
@@ -0,0 +1,156 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class ShrapnelInfo : IProjectileInfo
+	{
+		[Desc("Projectile speed in WDist / tick, two values indicate variable velocity.")]
+		public readonly WDist[] Speed = { new WDist(17) };
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
```

Comment:
This should be determined by the ThrowsShrapnel trait. The cohesion here is OK imo, as this is a case-specific optimization.